cranelift/codegen/src/isa/aarch64/inst.isle

```diff
@@ -1289,6 +1289,12 @@
             (Size64 64)
             (Size128 128)))
 
+;; Difference (32 - ty), useful for narrow calculations with 32-bit
+;; instructions.
+(decl diff_from_32 (Type) u8)
+(rule (diff_from_32 $I8) 24)
+(rule (diff_from_32 $I16) 16)
```

Comment:
I added this because this same pattern is used enough that a helper seems warranted (`cls`, `clz`, `bitrev`) but saving changing those for another PR.